src/GraphQL/API.hs

```diff
@@ -282,6 +291,53 @@ instance forall t. (HasAnnotatedInputType t) => HasAnnotatedInputType (List t) w
 instance forall ks enum. (KnownSymbol ks, GraphQLEnum enum) => HasAnnotatedInputType (Enum ks enum) where
   getAnnotatedInputType = do
     let name = nameFromSymbol @ks
-        enums = sequenceA (enumValues @enum Proxy) :: Either NameError [Name]
+        enums = sequenceA (enumValues @enum) :: Either NameError [Name]
     let et = EnumTypeDefinition <$> name <*> map (map EnumValueDefinition) enums
     TypeNonNull . NonNullTypeNamed . DefinedInputType . InputTypeDefinitionEnum <$> et
+
+
+-- Generic getAnnotatedInputType function
+class GenericAnnotatedInputType (f :: Type -> Type) where
+  genericGetAnnotatedInputType :: Either NameError (AnnotatedType InputType)
+
+class GenericInputObjectFieldDefinitions (f :: Type -> Type) where
+  genericGetInputObjectFieldDefinitions :: Either NameError [InputObjectFieldDefinition]
+
+instance forall dataName consName records s l p.
+  ( KnownSymbol dataName
+  , KnownSymbol consName
+  , GenericInputObjectFieldDefinitions records
+  ) => GenericAnnotatedInputType (D1 ('MetaData dataName s l 'False)
+                                  (C1 ('MetaCons consName p 'True) records
+                                  )) where
+  genericGetAnnotatedInputType = do
+    name <- nameFromSymbol @dataName
```

Comment:
We could do that but I'm not sure what it buys us: We need to return an error anyway for `HasAnnotatedType`. I.e. what do you suggest we do instead of the current implementation?